src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -686,13 +686,13 @@ private function processOnUpdatedHook(object $component, string $frontendName, L
             }
 
             $key = \sprintf('%s.%s', $frontendName, $propName);
-            if (!$dehydratedUpdatedProps->hasPropValue($key)) {
-                continue;
-            }
+            $fullPaths = $dehydratedUpdatedProps->searchFullPathsForProperty($key);
 
-            $this->ensureOnUpdatedMethodExists($component, $funcName);
-            $propertyOldValue = $dehydratedOriginalProps->getPropValue($key);
-            $component->{$funcName}($propertyOldValue);
+            foreach ($fullPaths as $fullPath) {
```

Comment:
This loop concern one callback for a prop. It should not be called multiple times. I this you could "just" replace the prevous "hasPropValue" to some "matchPropValue" or something like this and let the rest flow.